worker/pkg/workflows/datasync/activities/activities_test.go

```diff
@@ -310,6 +310,154 @@ func Test_buildProcessorMutation(t *testing.T) {
 	})
 	assert.Error(t, err)
 	assert.Empty(t, output)
+
+}
+
+func Test_buildProcessorConfigJavascript(t *testing.T) {
+	mockTransformerClient := mgmtv1alpha1connect.NewMockTransformersServiceClient(t)
+	mockJobClient := mgmtv1alpha1connect.NewMockJobServiceClient(t)
+	mockConnectionClient := mgmtv1alpha1connect.NewMockConnectionServiceClient(t)
+
+	pgcache := map[string]pg_queries.DBTX{
+		"fake-prod-url":  pg_queries.NewMockDBTX(t),
+		"fake-stage-url": pg_queries.NewMockDBTX(t),
+	}
+	pgquerier := pg_queries.NewMockQuerier(t)
+	mysqlcache := map[string]mysql_queries.DBTX{}
+	mysqlquerier := mysql_queries.NewMockQuerier(t)
+
+	bbuilder := newBenthosBuilder(pgcache, pgquerier, mysqlcache, mysqlquerier, mockJobClient, mockConnectionClient, mockTransformerClient)
+	ctx := context.Background()
+
+	mockTransformerClient.On(
+		"GetUserDefinedTransformerById",
+		mock.Anything,
+		connect.NewRequest(&mgmtv1alpha1.GetUserDefinedTransformerByIdRequest{
+			TransformerId: "456",
+		}),
+	).Return(connect.NewResponse(&mgmtv1alpha1.GetUserDefinedTransformerByIdResponse{
+		Transformer: &mgmtv1alpha1.UserDefinedTransformer{
+			Id:          "456",
+			Name:        "stage",
+			Description: "description",
+			DataType:    "string",
+			Source:      "javascript",
+			Config: &mgmtv1alpha1.TransformerConfig{
+				Config: &mgmtv1alpha1.TransformerConfig_JavascriptConfig{
+					JavascriptConfig: &mgmtv1alpha1.Javascript{
+						Code: `(()=>{var payload = benthos.v0_msg_as_structured();payload.value+=" helloee";})();`,
```

Comment:
I'm pretty confused by this as it's not really at all what we talked about.

The point was to abstract away the benthos stuff from the user, where as this is having them define everything.

Maybe we can talk about this more offline.